apps/speed.c

```diff
@@ -3483,10 +3484,11 @@ static int do_multi(int multi, int size_num)
         } else {
             close(fd[0]);
             close(1);
-            if (dup(fd[1]) == -1) {
+            if ((dup_fd = dup(fd[1])) == -1) {
```

Comment:
This is definitely incorrect.  This is duplicating the FD to be standard out, it doesn't need to be closed immediately.

Is this from a static analyser?  If so, it's a false positive.

Reply:
Hm, `dup()` use a new file descriptor. Closing the useless file descriptor in right time is more better. no?

Reply:
The file descriptor isn't useless and it isn't a resource leak.

In this case `dup()` will return 1.  That's _stdout_.  By closing it immediately afterwards, _stdout_ will be closed.  This is pretty standard `fork()` processing.

Reply:
> By closing it immediately afterwards, stdout will be closed.
I am not sure about it. I remebered that file descripter have ref count to count the number of file descriptors.

> This is pretty standard fork() processing.
You are right. If this `fork()` processing finished, this fd will be closed~.

Reply:
The int descriptor cannot possibly be reference counted.  How many copies of '1' does a program contain, how many are a file descriptor?  The kernel doesn't know.

Reply:
Thanks, make sense :)

Reply:
After double check, I see your points. I closed it since it's not a bug. Thanks for reviewing.